src/main.ts

```diff
@@ -11,50 +10,34 @@ import { chmod } from 'fs/promises'
 export async function run(): Promise<void> {
   try {
     const pklVersion = core.getInput('pkl-version')
-    const githubToken = core.getInput('github-token')
+    core.debug(`Installing Pkl version ${pklVersion}`)
 
-    if (!githubToken) {
-      throw new Error('Input "github-token" is required')
-    }
+    // Try to retrieve from cache first
+    let cachedPath = tc.find('pkl', pklVersion)
```

Comment:
We should probably add the optional `arch` parameter: https://github.com/actions/toolkit/blob/main/packages/tool-cache/src/tool-cache.ts#L498
Otherwise macOS builds (which have different architectures for the same OS and Pkl version) might clobber each other.

Reply:
Good catch! Will do

Reply:
On a second look, it appears it already defaults to the current arch: https://github.com/actions/toolkit/blob/6c4e082c181a51609197e536ef5255a0c9baeef7/packages/tool-cache/src/tool-cache.ts#L493

Reply:
nice find 👍